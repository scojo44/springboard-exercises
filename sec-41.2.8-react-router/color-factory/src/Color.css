.Color {
  height: 100%;
  font-size: 15vw;
  margin: 0;
  padding: 1rem;
  text-align: center;
}
