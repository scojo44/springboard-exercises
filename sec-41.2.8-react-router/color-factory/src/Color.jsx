import React from "react"
import {useParams, Navigate} from 'react-router-dom'
import './Color.css'

function ColorList({colors}) {
  const color = colors.find(c => c.name === useParams().color);

  if(!color) return <Navigate to="/colors" />;

  return (
    <h2 className="Color" style={{backgroundColor: color.value}}>
      {color.name}
    </h2>
  );
}

export default ColorList
