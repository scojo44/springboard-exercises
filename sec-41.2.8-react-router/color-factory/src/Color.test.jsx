import {render} from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import Color from './Color'

const TEST_COLORS = [
  {
    id: 1,
    name: 'purple',
    value: '#FF00FF'
  }
]

vi.mock('react-router-dom', async () => {
  return {
    ...await vi.importActual('react-router-dom'),
    useParams: () => ({color: 'purple'})
  };
});

describe('Color Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <Color colors={TEST_COLORS} />
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <Color colors={TEST_COLORS} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
