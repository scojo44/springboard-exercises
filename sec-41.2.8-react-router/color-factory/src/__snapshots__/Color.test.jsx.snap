// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Color Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <p
    class="Color"
    style="background-color: rgb(255, 0, 255);"
  >
    purple
  </p>
</DocumentFragment>
`;
